shroud/typemap.py

```diff
@@ -145,6 +157,8 @@ def __init__(self, name, **kw):
         #            setattr(self, key, defvalue)
         self.__dict__.update(self.defaults)  # set all default values
         self.update(kw)
+        if not self.flat_name:
+            self.flat_name = flatten_name(self.cxx_type or "JJJTypemap")
```

Comment:
``JJJTypemap`` ?

Reply:
During testing cxx_type was None for some types.   JJJTypemap will then show up in the output helping me identify how that type is being used.  So it's more of a debugging aid.

Reply:
I tracked down why cxx_type was None in some cases.  I committed fixes which properly sets flat_name for typedefs, enums, and template parameters.

Reply:
I figured it was a debugging artifact. Thanks for looking into it.